src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentTypeFinder_DerivedClasses.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.FindSymbols
+{
+    using SymbolSet = HashSet<INamedTypeSymbol>;
+
+    internal static partial class DependentTypeFinder
+    {
+        /// <summary>
+        /// This is an internal implementation of <see cref="SymbolFinder.FindDerivedClassesAsync(INamedTypeSymbol, Solution, IImmutableSet{Project}, CancellationToken)"/>, which is a publically callable method.
+        /// </summary>
+        public static Task<ImmutableArray<INamedTypeSymbol>> FindAndCacheDerivedClassesAsync(
```

Comment:
renamed methods for clarity.  tehre are two forms now (and will be 3 once OOP is involved in teh upcoming PR).  There are the FindAndCacheXXX methods, and the FindXXX methods.  THe former go through the entire caching system here, deferring to the latter to actually perform the work if the values are not cached.